avaframe/com1DFAPy/timeDiscretizations.py

```diff
@@ -0,0 +1,50 @@
+"""
+    function regarding time discretization and time stepping for com1DFA
+
+    This file is part of Avaframe.
+"""
+
+# Load modules
+import os
+import glob
+import logging
+import numpy as np
+
+# Local imports
+import avaframe.in3Utils.fileHandlerUtils as fU
+import avaframe.in2Trans.ascUtils as IOf
+from avaframe.in3Utils import cfgUtils
+from avaframe.in3Utils import logUtils
+
+
+# create local logger
+# change log level in calling module to DEBUG to see log messages
+log = logging.getLogger(__name__)
+
+
+def getTimeStep(particles, dem, cfg):
+    """ Compute cfl time step  """
+
+    # determine max velocity of particles
+    vmagnitude = np.sqrt(particles['ux']**2 + particles['uy']**2 + particles['uz']**2)
+    vmax = np.amax(vmagnitude)
+
+    # get cell size
+    csz = dem['header'].cellsize
+
+    # courant number
+    cMax = float(cfg['cMax'])
+
+    # compute stable time step
+    # if velocity is zero - divided by zero error so to avoid:
+    if vmax == 0.0:
+        dtStable = float(cfg['maxdT'])
+    else:
+        dtStable = (cMax * csz) / vmax
+
```

Comment:
So this could lead to an extremely small timestep which would basically grind the computations to a halt. Should we implement some checking, like if the timestep is below a certain limit, we try with a less conservative courant number up to a certain point?
Or does this make it way to overcomplicated for now?

Reply:
The way it is now if the velocity is zero and hence no time step can be computed according to the function, it depends on what you set in the .ini file as maxdT. Which is where at the moment the time step that is used is read. I would implement the suggestion in the future in case we need the cfl time step for the computations.